### PR TITLE
bf: S3C-4358 support versioned object lock requests

### PR DESCRIPTION
**Pull request by ilkescality**

### CODE REVIEW

**Comment by rahulreddy** [comment]
Fixed using `rm -rf node_modules && yarn cache clean && yarn`

**Review by tmacro — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

File: tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js [hunk 1]
```diff
@@ -18,17 +18,26 @@ const mockLegalHold = {
...
     if (!status) {
-        return { Bucket: bucket, Key: key };
+        return params;
     }
-    return {
-        Bucket: bucket,
-        Key: key,
+    if (!versionId) {
+        Object.assign(params, {
+            VersionId: versionId,
```
Comment (jonathan-gramain) [inline tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js:28]: Isn't the logic reversed here? if `versionId` is false-ish it's likely undefined.

Comment (jonathan-gramain) [reply tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js:28]: I mean the logic of the `if`

Comment (ilkescality) [reply tests/functional/aws-node-sdk/test/object/putObjectLegalHold.js:28]: it is -- supposed to be if `versionId` not `!versionId` just fixed it should be good now.

**Review by jonathan-gramain — APPROVED** [review]

**Review by saitharunthodupunuru — APPROVED** [review]

**Pull request merged**